src/matrix/mod.rs

```diff
@@ -84,41 +83,59 @@ impl MatrixClient {
         base_url: Url,
         user: String,
         password: String,
-    ) ->  Result<MatrixClient, Error> {
-        let host = base_url
+    ) -> Result<MatrixClient, Error> {
+        let _host = base_url
             .host_str()
             .expect("expected host in base_url")
             .to_string();
-        let port = base_url.port_or_known_default().unwrap();
-        let tls = match base_url.scheme() {
+        let _port = base_url.port_or_known_default().unwrap();
+        let _tls = match base_url.scheme() {
             "http" => false,
             "https" => true,
             _ => panic!("Unrecognized scheme {}", base_url.scheme()),
         };
```

Comment:
previously these were used with the custom http implementation, but since `url::Url` and is later converted to `hyper::Uri`, i think these are safe to remove.

Reply:
Yeah, sounds good.